api/src/dto/ChangeDto.ts

```diff
@@ -0,0 +1,21 @@
+import { DocType, Uuid } from "../types";
+import { IsNotEmpty, IsObject } from "class-validator";
+import { GroupAclEntryDto } from "./GroupAclEntryDto";
+
+/**
+ * Database structured Post object
+ */
+export class ChangeDto {
+    @IsNotEmpty()
+    _id: Uuid;
+    @IsNotEmpty()
+    type: DocType.Change;
+    @IsNotEmpty()
+    docId: Uuid;
+    @IsNotEmpty()
+    docType: DocType;
+    memberOf?: Uuid[];
```

Comment:
How would we handle advanced validation rules? Just thinking that here we specifically want to check if the passed Uuid's are valid group id's and if the user has access to them. Or do you think we should handle that in the API code and just discard any non-valid id's?

Reply:
Can add multiple validation decorations. Access validation will be done separately in the API by making calls to the permission system